### PR TITLE
src/Makefile, generate_makefile: Fixes for make clean

### PR DESCRIPTION
**Pull request by ndellingwood**
Updated so that calling 'make clean' works without errors.

### CODE REVIEW

**Comment by ndellingwood** [comment]
PR to potentially address issue #353

File: src/Makefile [hunk 1]
```diff
@@ -93,3 +93,7 @@ build: kokkoskernels-build-lib
 
 install: kokkoskernels-install
  
+clean: kokkoskernels-clean kokkos-clean
```
Comment (kyungjoo-kim) [inline src/Makefile]: Do we also need to clean kokkos ?

Comment (ndellingwood) [reply src/Makefile]: @kyungjoo-kim maybe not, I had put that in there while I was making sense of the issue and did not retest without once I had a better idea what was going on. Would you like me to test without and update the PR?

Comment (ndellingwood) [reply src/Makefile]: Sounds good, thanks @kyungjoo-kim !

**Pull request merged**